vim-engine/src/main/kotlin/com/maddyhome/idea/vim/action/ex/LeaveCommandLineAction.kt

```diff
@@ -1,35 +1,30 @@
 /*
- * Copyright 2003-2023 The IdeaVim authors
+ * Copyright 2003-2024 The IdeaVim authors
  *
  * Use of this source code is governed by an MIT-style
  * license that can be found in the LICENSE.txt file or at
  * https://opensource.org/licenses/MIT.
  */
+
 package com.maddyhome.idea.vim.action.ex
 
 import com.intellij.vim.annotations.CommandOrMotion
 import com.intellij.vim.annotations.Mode
-import com.maddyhome.idea.vim.VimPlugin
 import com.maddyhome.idea.vim.api.ExecutionContext
 import com.maddyhome.idea.vim.api.VimEditor
 import com.maddyhome.idea.vim.command.Command
 import com.maddyhome.idea.vim.command.CommandFlags
 import com.maddyhome.idea.vim.command.OperatorArguments
 import com.maddyhome.idea.vim.handler.VimActionHandler
+import com.maddyhome.idea.vim.helper.enumSetOf
 import java.util.*
 
-/**
- * Called by KeyHandler to process the contents of the ex entry panel
- *
- * The mapping for this action means that the ex command is executed as a write action
- */
-@CommandOrMotion(keys = ["<CR>", "<C-M>", "<C-J>"], modes = [Mode.CMD_LINE])
-public class ProcessExEntryAction : VimActionHandler.SingleExecution() {
-  override val type: Command.Type = Command.Type.OTHER_SELF_SYNCHRONIZED
-
-  override val flags: EnumSet<CommandFlags> = EnumSet.of(CommandFlags.FLAG_COMPLETE_EX)
+@CommandOrMotion(keys = ["<Esc>", "<C-[>", "<C-C>"], modes = [Mode.CMD_LINE])
+public class LeaveCommandLineAction : VimActionHandler.SingleExecution() {
```

Comment:
How does this command work if it just returns `true`?

Reply:
This command has `CommandFlags.FLAG_END_EX` which is handled in command consumer (it closes command line, resets commandBuilder etc) 
I don't like idea of having this flag or this logic in `CommandConsumer`, however it's not the only command with `CommandFlags.FLAG_END_EX` and some of such commands are `ForEachCaret`. This "end ex" code should be executed only once and can't be moved to commands body (at least before we add some `preExecute` method for commands that will be executed once`)